v4/export/config.go

```diff
@@ -28,28 +33,85 @@ func DefaultConfig() *Config {
 		Logger:        &DummyLogger{},
 		FileSize:      UnspecifiedSize,
 		OutputDirPath: ".",
+		ServerInfo:    UnknownServerInfo,
+		SortByPk:      false,
 	}
 }
 
 func (conf *Config) getDSN(db string) string {
 	return fmt.Sprintf("%s:%s@tcp(%s:%d)/%s", conf.User, conf.Password, conf.Host, conf.Port, db)
 }
 
-func extractOutputConfig(conf *Config) *Config {
-	return &Config{
-		Logger:        conf.Logger,
-		FileSize:      conf.FileSize,
-		OutputDirPath: conf.OutputDirPath,
+const UnspecifiedSize = 0
+
+type ServerInfo struct {
+	ServerType    ServerType
+	ServerVersion *ServerVersion
```

Comment:
Let's use a semver library to store the server version (TiDB uses `github.com/coreos/go-semver`). So we don't need to repeat that "parse `1.2.3` into `[1, 2, 3]`" code.